tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,463 @@
+
+			By("verifying pod count as 1")
+			podCount := ops.GetPodRunningCountEventually(namespaceObj.Name, label, 1)
+			Expect(podCount).To(Equal(1), "while verifying pod count")
+		})
+	})
+
+	When("writing data more than quota limit into the hostpath volume", func() {
+		It("should not be able to write more than the enforced limit", func() {
+			By("Verifying the quota applied on the volume works")
+			execOption := NewOptions()
+			// The command formed below is attempting to create a file named “test.txt” inside the attached volume,
+			// its size increasing 1M in each iteration of the loop should show us the quota in action.
+			option := execOption.WithPodName(createdPod.Name).
+				WithContainer("busybox").
+				WithNamespace(namespaceObj.Name).
+				WithCommand([]string{"/bin/sh", "-c", fmt.Sprintf(`for i in $(seq 1 7); do dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=$i 2>/dev/null; if [ $? -ne 0 ]; then echo "filesize of $i Mb failed, quota reached!"; else echo "filesize of $i Mb was ok"; fi; done`)}...)
+			stdOut, stdErr, err := ops.ExecPod(option)
+			fmt.Printf("When running command to test enforced quota. stdOut: {%s}, stderr: {%s}, error: {%v}", stdOut, stdErr, err)
+			Expect(stdErr).NotTo(BeNil(), "trying to write beyond the quota limit should not be allowed")
+			Expect(stdOut).Should(ContainSubstring("filesize of 4 Mb was ok"), "trying to write till the quota limit should be allowed")
+			Expect(stdOut).Should(ContainSubstring("filesize of 5 Mb failed, quota reached!"), "trying to write beyond the quota limit should not be allowed")
+		})
+	})
+
+	When("Pod consuming pvc "+pvcName+" is deleted", func() {
+		It("should delete the pod", func() {
+			By("deleting above pod")
+			err = ops.PodClient.WithNamespace(namespaceObj.Name).Delete(context.TODO(), podName, &metav1.DeleteOptions{})
+			Expect(err).To(
+				BeNil(),
+				"while deleting pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("verifying pod count as 0")
+			podCount := ops.GetPodRunningCountEventually(namespaceObj.Name, label, 0)
+			Expect(podCount).To(Equal(0), "while verifying pod count")
+		})
+	})
+
+	When("pvc with storageclass "+scName+" is deleted", func() {
+		It("should delete the pvc", func() {
+			By("deleting the PVC with StorageClass " + scName)
+			pvName := ops.GetPVNameFromPVCName(pvcName)
+			Expect(pvName).ToNot(
+				BeEmpty(),
+				"while getting Spec.VolumeName from "+
+					"PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("deleting above PVC")
+			err = ops.PVCClient.Delete(context.TODO(), pvcName, &metav1.DeleteOptions{})
+			Expect(err).To(
+				BeNil(),
+				"while deleting pvc {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("verifying PVC is deleted")
+			status := ops.IsPVCDeletedEventually(pvcName, namespaceObj.Name)
+			Expect(status).To(
+				BeTrue(),
+				"when checking status of deleted PVC {%s}",
+				pvcName,
+			)
+		})
+	})
```

Comment:
Here, also StorageClass created for test needs to be deleted

Reply:
same comment as above.

Reply:
Creating at one place and deleting at a different places is not a good thing... But it is ok if current test code is doing the same